src/JuMP.jl

```diff
@@ -738,5 +729,19 @@ include("print.jl")
 # Nonlinear-specific code
 include("nlp.jl")
 
+
+typealias _getTypes Union{NonlinearExpression,NonlinearParameter, AffExpr}
+getValue{T<:_getTypes}(arr::Array{T}) = map(getValue, arr)
```

Comment:
Are you sure? ``JuMPTypes`` appears a lot in the operator overload code, but for scalar ops it looks like the signatures are of the form ``{C,V<:JuMPTypes}(lhs::V, rhs::GenericAffExpr{C,V})``, which is already nonsense if ``V==NonlinearExpression``. Then the vectorized ops all fall back on these scalar ops.